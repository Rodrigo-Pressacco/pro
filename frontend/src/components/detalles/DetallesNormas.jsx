import React from 'react';
import Normas from '../../utils/Normas.json'
import "../../styles/detalles/DetallesCaracteristicas.css";


function ElementosNormas(props) {
    return (
        <li>
            <p> {props.element.norma} </p>
        </li>
    )
}

function ElementosSys (props) {
    return (
        <li>
            <p> {props.element.sys} </p>
        </li>
    )
}


const DetallesNormas = () => {

    return (

        <div className="div-detalles-caracteristicas">
            <div className="div-detalles-h3">
                <h3>Que tenés que saber</h3>
            </div>
                <div className="columnaNormas">
                    <div className="columnax3"> 
                        <h4 className="tituloNorma">Normas de la Casa</h4>
                            {Normas.map((element) => (<ElementosNormas key={element.id} element={element}/>))}
                    </div>

                    <div className="columnax3">
                        <h4 className="tituloNorma">Salud y seguridad</h4>
                            {Normas.map((element) => (<ElementosSys key={element.id} element={element}/>))}
                    </div>
                
                    <div className="columnax3">
                        <h4 className="tituloNorma">Política de cancelación</h4>
                            <p className="textoNorma textoNormaEspecifico"> Agregá las fechas de tu viaje para obtener los detalles de cancelación de esta estadía.</p>
                    </div>
                </div>
        </div>
    )
}


export default DetallesNormas;