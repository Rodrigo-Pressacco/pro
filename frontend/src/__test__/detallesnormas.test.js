import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import DetallesNormas from '../components/detalles/DetallesNormas'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<DetallesNormas />', () => {
    let normas
    let ElementosNormas = jest.fn()
    let ElementosSys = jest.fn()

    beforeEach(() => {
        normas = shallow(<DetallesNormas ElementosNormas={ElementosNormas} ElementosSys={ElementosSys} />)
    });

    it("Deberia mostrar <DetallesNormas/> correctamente", () => {
        expect(normas).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta h3", () => {
        let node = normas.find("h3")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 3 etiquetas h4", () => {
        let node = normas.find("h4")
        expect(node.length).toEqual(3)
    });

    it("Deberia retornar 1 etiqueta p", () => {
        let node = normas.find("p")
        expect(node.length).toEqual(1)
    });

})