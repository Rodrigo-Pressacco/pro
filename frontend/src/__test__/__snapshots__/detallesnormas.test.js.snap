// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetallesNormas /> Deberia mostrar <DetallesNormas/> correctamente 1`] = `
<div
  className="div-detalles-caracteristicas"
>
  <div
    className="div-detalles-h3"
  >
    <h3>
      Que tenés que saber
    </h3>
  </div>
  <div
    className="columnaNormas"
  >
    <div
      className="columnax3"
    >
      <h4
        className="tituloNorma"
      >
        Normas de la Casa
      </h4>
      <ElementosNormas
        element={
          Object {
            "id": "1",
            "norma": "No se permiten mascotas",
            "sys": "Se siguen las normas de bioseguridad",
          }
        }
        key="1"
      />
      <ElementosNormas
        element={
          Object {
            "id": "2",
            "norma": "Check - out 3:00 p.m",
            "sys": "Detectores de humo",
          }
        }
        key="2"
      />
      <ElementosNormas
        element={
          Object {
            "id": "3",
            "norma": "No se permite fumar",
            "sys": "Se aplican pautas de distanciamiento",
          }
        }
        key="3"
      />
      <ElementosNormas
        element={
          Object {
            "id": "4",
            "norma": "No se permiten fiestas",
            "sys": "Extintores en los pasillos",
          }
        }
        key="4"
      />
    </div>
    <div
      className="columnax3"
    >
      <h4
        className="tituloNorma"
      >
        Salud y seguridad
      </h4>
      <ElementosSys
        element={
          Object {
            "id": "1",
            "norma": "No se permiten mascotas",
            "sys": "Se siguen las normas de bioseguridad",
          }
        }
        key="1"
      />
      <ElementosSys
        element={
          Object {
            "id": "2",
            "norma": "Check - out 3:00 p.m",
            "sys": "Detectores de humo",
          }
        }
        key="2"
      />
      <ElementosSys
        element={
          Object {
            "id": "3",
            "norma": "No se permite fumar",
            "sys": "Se aplican pautas de distanciamiento",
          }
        }
        key="3"
      />
      <ElementosSys
        element={
          Object {
            "id": "4",
            "norma": "No se permiten fiestas",
            "sys": "Extintores en los pasillos",
          }
        }
        key="4"
      />
    </div>
    <div
      className="columnax3"
    >
      <h4
        className="tituloNorma"
      >
        Política de cancelación
      </h4>
      <p
        className="textoNorma textoNormaEspecifico"
      >
         Agregá las fechas de tu viaje para obtener los detalles de cancelación de esta estadía.
      </p>
    </div>
  </div>
</div>
`;
